app/components/progressive_image/progressive_image.js

```diff
@@ -86,12 +87,17 @@ export default class ProgressiveImage extends PureComponent {
 
     setThumbnail = (thumb) => {
         if (this.subscribedToCache) {
-            const {filename, imageUri} = this.props;
-            this.setState({thumb}, () => {
-                setTimeout(() => {
-                    ImageCacheManager.cache(filename, imageUri, this.setImage);
-                }, 300);
-            });
+            if (!thumb && !this.state.failedImageLoad) {
+                this.load();
+                this.setState({failedImageLoad: true});
+            } else {
+                const {filename, imageUri} = this.props;
+                this.setState({thumb}, () => {
+                    setTimeout(() => {
+                        ImageCacheManager.cache(filename, imageUri, this.setImage);
+                    }, 300);
+                });
+            }
```

Comment:
I'm missing a case here where it failed to load the second time, we should probably have some sort of image broken thumbnail to display, so the user is aware that something went wrong.

Reply:
I think that's already happening here, except it's currently showing a default image. We could just change the default image to a more 'broken' looking image.

Reply:
@Willyfrog Ya i think we should have a separate ticket for this. Probably even communicate that we are fetching the image as well. Seems odd that there is not much of indication about it.

Reply:
@devinbinnie I saw the default, but not sure if it would convey that meaning

@sudheerDev done: https://mattermost.atlassian.net/browse/MM-19522 waiting for grooming :)

Reply:
@imisshtml I meant test case for the retry `this.load();`.

Reply:
ohhhh my, sorry I misunderstood, will fix that.